### PR TITLE
cspl-1139: Make Probe timers dynamic

### PR DESCRIPTION
**Pull request by sgontla**
- Logic to dynamically adjust the Readiness and Liveness initial delay times to accommodate the app installations
- config option to specific the Liveness and Readiness Probe initial delay time

### CODE REVIEW

**Comment by sgontla** [comment]
Just a note, dynamic delay calculation may need further tuning, based on our observations going forward, but the logic is not going to change much.

**Review by jryb — APPROVED** [review]

File: pkg/splunk/enterprise/configuration.go [hunk 1]
```diff
@@ -748,6 +736,53 @@ func updateSplunkPodTemplateWithConfig(client splcommon.ControllerClient, podTem
...
 }
 
+// getLivenessProbe the probe for checking the liveness of the Pod
+// uses script provided by enterprise container to check if pod is alive
+func getLivenessProbe(spec *enterprisev1.CommonSplunkSpec, additionalDelay int32) *corev1.Probe {
+	livenessDelay := livenessProbeDefaultDelay + additionalDelay
+	if spec.LivenessInitialDelaySeconds > livenessDelay {
+		livenessDelay = spec.LivenessInitialDelaySeconds
+	}
+
```
Comment (gaurav-splunk) [inline pkg/splunk/enterprise/configuration.go:746]: probably a log msg here saying what is the value of livenessDelay. Might be helpful later while debugging issues related to this.

File: pkg/splunk/enterprise/configuration.go [hunk 2]
```diff
@@ -748,6 +736,53 @@ func updateSplunkPodTemplateWithConfig(client splcommon.ControllerClient, podTem
...
+// getReadinessProbe provides the probe for checking the readiness of the Pod
+// pod is ready if container artifact file is created with contents of "started".
+// this indicates that all the the ansible plays executed at startup have completed.
+func getReadinessProbe(spec *enterprisev1.CommonSplunkSpec, additionalDelay int32) *corev1.Probe {
+	readinessDelay := readinessProbeDefaultDelay + additionalDelay
+	if spec.ReadinessInitialDelaySeconds > readinessDelay {
+		readinessDelay = spec.ReadinessInitialDelaySeconds
+	}
+
```
Comment (gaurav-splunk) [inline pkg/splunk/enterprise/configuration.go:762]: ditto

File: pkg/splunk/enterprise/util_test.go [hunk 3]
```diff
@@ -770,9 +770,9 @@ func TestSetLastAppInfoCheckTime(t *testing.T) {
 }
 
 func TestGetNextRequeueTime(t *testing.T) {
-	appFrameworkContext := enterprisev1.AppDeploymentContext{}
-	appFrameworkContext.AppsRepoStatusPollInterval = 60
-	nextRequeueTime := GetNextRequeueTime(appFrameworkContext.AppsRepoStatusPollInterval, (time.Now().Unix() - int64(40)))
+	appFrameworkRef := enterprisev1.AppFrameworkSpec{}
+	appFrameworkRef.AppsRepoPollInterval = 60
+	nextRequeueTime := GetNextRequeueTime(appFrameworkRef.AppsRepoPollInterval, (time.Now().Unix() - int64(40)))
```
[the remark this answers is not in the document]
Comment (sgontla) [reply pkg/splunk/enterprise/util_test.go:775]: Agree

**Review by gaurav-splunk — APPROVED** [review]

**Review by smohan-splunk — APPROVED** [review]

**Pull request merged**